compiler/src/main/kotlin/dev/zacsweers/metro/compiler/ir/ir.kt

```diff
@@ -624,7 +624,7 @@ internal fun assignConstructorParamsToFields(
       val irField =
         clazz
           .addField(
-            toJavaIdentifier(irParameter.name.asString()),
+            toValidatedIdentifier(irParameter.name.asString()),
```

Comment:
note to self: this should probably ask for a `NameAllocator` from somewhere rather than calling this API directly